fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java

```diff
@@ -652,16 +644,31 @@ public String getCurrentConnectedFEIp() {
         return currentConnectedFEIp;
     }
 
+    /**
+     * Keep it just for directly controlling kill_thread_timeout in {@link #checkTimeout(long)}
+     */
     public void setExecTimeout(int timeout) {
         executionTimeoutS = timeout;
     }
 
-    public long resetExecTimeoutByInsert() {
-        executionTimeoutS = Math.max(executionTimeoutS, sessionVariable.getInsertTimeoutS());
-        return executionTimeoutS;
-    }
-
+    /**
+     * We calculate and get the exact execution timeout here, rather than setting
+     * execution timeout in many other places.
+     *
+     * @return exact execution timeout
+     */
     public int getExecTimeout() {
+        userQueryTimeout = Env.getCurrentEnv().getAuth().getQueryTimeout(qualifiedUser);
```

Comment:
this is a temp value of uer timeout.  If user timeout was -1, it will make the killer thread bad.

In my suggestion, do not modify the value of executionTimeoutS in `getExecTimeout()` just return the right value.

Reply:
If user timeout was -1, we will not use it as execution timeout, and will use session var instead. And in ConnectContext#checkTimeout, we use directly use executionTimeout for kill thread timeout checking. If we can change the setExecTimeout behavior in MysqlLoadManager#executeMySqlLoadJobFromStmt, the problem will be completely solved.